src/test/java/fi/aalto/cs/apluscourses/presentation/base/SelectableNodeViewModelTest.java

```diff
@@ -94,4 +94,23 @@ public void testApplyFilterReturnsEmpty() throws InterruptedException {
     assertFalse(result.isPresent());
     assertTrue(node.isVisible());
   }
+
+  @Test
+  public void testSetVisibilityByFilterResult() throws InterruptedException {
+    Filter filter = mock(Filter.class, new Returns(Optional.empty()));
+
+    TestNodeViewModel nodeHide =
+        new TestNodeViewModel(4, new Object(), Collections.emptyList(), true);
+
+    Optional<Boolean> resultHide = nodeHide.applyFilter(filter);
+    assertFalse(resultHide.isPresent());
+    assertFalse(nodeHide.isVisible());
+
+    TestNodeViewModel nodeShow =
+        new TestNodeViewModel(4, new Object(), Collections.emptyList(), false);
+
+    Optional<Boolean> resultShow = nodeHide.applyFilter(filter);
+    assertFalse(resultShow.isPresent());
+    assertTrue(nodeShow.isVisible());
+  }
```

Comment:
I'm afraid this still tests mostly `TestNodeViewModel` mockup. 😕 

Maybe delete this test or change it to something like (after removing override from `TestNodeViewModel`):

```
node.setVisibilityByFilterResult(Optional.of(false));
assertFalse(node.isVisible);
// and so on
```